content/blog/eng/operators-overview.md

```diff
@@ -0,0 +1,192 @@
+
+For example, in Perl coercion rules lead into 12 + "3.14" producing the result of 15.14. The text "3.14" is converted to the number 3.14 before addition can takes place. Further, 12 is an integer and 3.14 is either a floating or fixed-point number (a number that has a decimal place in it) so the integer is then converted to a floating point or fixed-point number respectively.
+```Perl
+$firstValue = 12;
+$secondValue = "3.14";
+print $firstValue + $secondValue; // 15.14
+```
+JavaScript follows opposite rules—finding the same expression above, it will convert the integer 12 into a string "12", then concatenate the two operands to form "123.14".
+```JavaScript
+let firstValue = 12;
+let secondValue = "3.14";
+print(firstValue+secondValue); // "123.14"
+```
+In the presence of coercions in a language, the programmer must be aware of the specific rules regarding operand types and the operation result type to avoid subtle programming mistakes.
+
+
+# Common operators
+
+## Access operator
+#### Symbol -> .
+
+**Access operator** is used for to access a member of a namespace or a type (type.member). For example we can use access operator to access a nested namespace  String.Indices or to access type members static and non-static.
+
+## Scope resolution operator
+#### Symbol -> ::
+The **scope resolution operator**, helps to identify and specify the context which an identifier refers.
+
+## Indexer operator
+#### Symbol -> [ ]
+The **indexer operator** is typically used to access a variable in a sequence or in a pointer element access.
+
+## Null-conditional operators
+#### Symbols -> ?. , ?[ ]
+**Null-conditional** operators allow access to a member only if that operand evaluates to non-null; otherwise, it returns null.
+
+## Compound Assignment Operators
+#### Symbols -> += , -=, *=, /=, %=, &=, |=, ^=, ...
+**Compound assignment operators** provide a shorter syntax for assigning the result for an arithmetic and bitwise operator. They perform the operation on the two operands before assigning the result to the first operand.
+For example  x += 2  is a shortcut for  x = x + 2.
+
+## Invocation operator
+#### Symbol -> ()
+The invocation operator () call a method or invoke a delegate.
+
+## Relational operator
+The relational operator are constructs that define a type of relationship between two entities, this definition includes both equality and inequality.
+
+An expression built with relational operators is called conditional expression and expressions of this type are defined **predicates**. If the language support the Boolean type (es. Java, PHP) the relational operator returns a true/false value, instead in other languages like C the relational operator returns 0 or 1, where the 0 correspond to false value and 1 to the true value.
+
+In some dynamically typed languages, equality can also be done at the type level, for example:
```

Comment:
"equality can also be done". I'm not sure about the verb to do, it sounds a bit too general.  
Equality can also be **expressed**?

Reply:
I replaced that sentence with this, what do you think?
`In some dynamic languages, we can match two values based on their types`

Reply:
I think it's ok!